sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1675,6 +1675,13 @@ class Analyzer(override val catalogManager: CatalogManager)
       // Skip the having clause here, this will be handled in ResolveAggregateFunctions.
       case h: UnresolvedHaving => h
 
+      case d @ DescribeColumn(rt: ResolvedTable, _, _) =>
+        rt.table match {
+          // References for v1 tables are resolved in DescribeColumnCommand.
+          case _: V1Table => d
```

Comment:
Shall we change the v1 command to take resolved `Attribute` directly?  Then we don't need the hack here.

This also reminds me that, for all v1 commands, we will lookup the table twice: once in the framework, and once inside the v1 command. This is kind of a perf regression: previously we simply lookup the catalog and fallback to v1 command if it's the session catalog, so the table lookup still happens once inside the v1 command. We can also update the v1 command and take the resolved table directly.

Reply:
Or, we don't care too much about the DDL perf, then I think resolving the column twice is also fine and we don't need this hack either.

Reply:
Since the v1 command supports views as well, updating v1 command to take resolved `Attribute` doesn't solve the issue completely. We have to pass `Attribute` and the original column name to v1 command (or have two different v1 commands that take either resolved attribute or column name). Do you still prefer resolving attributes here for v1 tables?

Reply:
Then we shouldn't change the v1 command.

Reply:
But seems like to resolve the column twice?

Reply:
We can re-construct the multi-part column name from `Attribute`: `attr.qualifier +: attr.name`